### DISCUSSION TITLE
Plotting grid

### DISCUSSION THREAD

**Discussion by vprigiob** [modflowpy/flopy#1639] [Q&A]
Hi,
I am trying to plot a grid of a model run with modflow 6.
I read the example at: https://github.com/modflowpy/flopy/blob/develop/examples/Notebooks/flopy3.3_PlotMapView.ipynb
I am using:

    sim = flopy.mf6.MFSimulation.load(sim_ws='.', load_only=['dis'], verbosity_level=0)
    mapview = flopy.plot.PlotMapView(model=sim)
    linecollection = mapview.plot_grid()

But it does not work.
I have also extracted the grid with the command line:
     mg = sim.get_model().modelgrid
and I have tried:
     linecollection = mg.plot_grid()
But plot_grid is not an attribute for modelgrid.

Could you please suggest how to extract the grid and plot it using flopy.mf6?
I could found an example in the github repository 

Thank you so much.

Best regards

**Comment by jlarsen-usgs** [comment]
@vprigiob 

You need to specify a model name in the `sim.get_model()` function call or it will return None. Two examples:

1) Using PlotMapView
```python
sim = flopy.mf6.MFSimulation.load(sim_ws='.', load_only=['dis'], verbosity_level=0)
mymodel = sim.get_model("my_model_name")
mapview = flopy.plot.PlotMapView(model=mymodel)
linecollection = mapview.plot_grid()
```

2) Using modelgrid.plot_grid()
```python
mymodel = sim.get_model("my_model_name")
mg = mymodel.modelgrid
linecollection = mg.plot_grid()
```

**Reply by vprigiob** [reply]
Hi,
Thank you for the suggestions.
Apparently, I cannot use PlotMapView because the model has a structured grid.

I am trying to use modelgrid.plot_grid(), instead. However,  when I try to plot the grid I get this message:
          raise NotImplementedError("must define grid_lines in child class")
      NotImplementedError: must define grid_lines in child class

As I am trying to plot the simulations results of an existing modflow model, I am wondering if
grid_lines relates to something to set in the model which is missing in my case.
Thank you for your help.
